modules/calendar/add.php

```diff
@@ -0,0 +1,548 @@
+      // $value is an array consisting of day type string followed by facility id string
+      // $prov_id is id string of a provider
+
+      $event_date = $date;  // event date
+      $event_end_date = "0000-00-00";  // Since, single event
+      // if $value[0] == "Full day"
+      $starttime = "{$GLOBALS['schedule_start']}:00:00";
+      $endtime = "{$GLOBALS['schedule_end']}:00:00";
+      if ($value[0] == "First half") {
+        $starttime = "{$GLOBALS['schedule_start']}:00:00";
+        $endtime = "13:00:00";
+      } elseif ($value[0] == "Second half") {
+        $starttime = "13:00:00";
+        $endtime = "{$GLOBALS['schedule_end']}:00:00";
+      }
+      $provider = $prov_id;
+
+      // Event location specifications
+      $locationspecs = array( "event_location" => "",
+                              "event_street1" => "",
+                              "event_street2" => "",
+                              "event_city" => "",
+                              "event_state" => "",
+                              "event_postal" => "" );
+      $locationspec = serialize($locationspecs);
+
+      // Recurring event specifications
+      $recurrspec = array( "event_repeat_freq" => "0",
+                           "event_repeat_freq_type" => "0",
+                           "event_repeat_on_num" => "1",
+                           "event_repeat_on_day" => "0",
+                           "event_repeat_on_freq" => "0",
+                           "exdate" => "" );
+
+      $args = array();
+        // specify values needed for the INSERT into table - libreehr_postcalendar_events - columns
+        $args['form_category'] = "6";  // Holiday
+        $args['new_multiple_value'] = "";
+        $args['form_provider'] = $provider;  // selected provider
+        $args['form_title'] = "Holidays";
+        $args['form_comments'] = "";
+        $args['event_date'] = $event_date;
+        $args['form_enddate'] = $event_end_date;
+        $args['duration'] = 0;  // no duration in provider events
+        $args['form_repeat'] = "0";  // Repeats box not checked
+        $args['recurrspec'] = $recurrspec;
+        $args['starttime'] = $starttime;
+        $args['endtime'] = $endtime;
+        $args['form_allday'] = 0;
+        $args['form_apptstatus'] = "-";  // None
+        $args['form_prefcat'] = "0";  // None
+        $args['locationspec'] = $locationspec;
+        // Taking facility & billing facility as selected facility of holiday
+        $args['facility'] = $value[1];
+        $args['billing_facility'] = $value[1];
+      // Insert events - A holiday event is made for each selected date,
+      // covering Calendar slots according to selected day type for that date with selected facility under all providers.
+      InsertEvent($args);
+    }
+
+    // reached here, means no conflicting dates, make requested events
+    foreach ($dateAndType as $date => $value) {
+      $providers = getProviderInfo();
```

Comment:
There is no facility check so the Holiday is applied to every facility, even if you select a facility it is still applied to all.

Reply:
@teryhill Facility id is stored in `$value` corresponding to `$date`. I tested with different facilities and holiday was made w.r.t to selected facilities. It wasn't applied to all, am I doing something wrong here?

Reply:
What is w.r.t?

Reply:
with respect to

Reply:
I have a provider that is only at one facility and other providers at another facility. If I select the facility that has one provider and make a holiday it makes it for "All" providers no matter which facility is selected.

Reply:
Oh, so you are talking about it being applied to all providers when we make a holiday event. That's the default behavior because facility is closed no matter which provider.

Reply:
What if they use this to close a facility because the facility is being repaired?

Reply:
To add under only those providers which are at that facility selected for holiday, I think we have to take help of getUsersFacility.

Reply:
@teryhill Will change adding under all providers to under only those providers specific to that selected facility.